R/positron.R

```diff
@@ -0,0 +1,89 @@
+ark_register_methods <- function() {
+  # TODO: Actually bump this to 0.1.176 after Ark PR merges and we bump the version
+  if (!ark_has_version("0.1.174")) {
+    return()
+  }
+
+  tryCatch(
+    ark_register_methods_impl(),
+    error = function(err) {
+      cli::cli_warn("Failed to register Ark methods.", parent = err)
+    }
+  )
+}
+
+ark_register_methods_impl <- function() {
+  ark_register_method(
+    "ark_positron_variable_display_value",
+    "duckplyr_df",
+    function(x, ...) {
+      duckplyr_df_variable_display_value(x)
+    }
+  )
+
+  ark_register_method(
+    "ark_positron_variable_display_type",
+    "duckplyr_df",
+    function(x, ...) {
+      duckplyr_df_variable_display_type(x)
+    }
+  )
+}
+
+ark_register_method <- function(generic, class, method) {
+  ARK_REGISTER_METHOD_FUNCTION <- ".ark.register_method"
+
+  if (!exists(ARK_REGISTER_METHOD_FUNCTION, mode = "function")) {
+    return()
+  }
+
+  call <- call(
+    ARK_REGISTER_METHOD_FUNCTION,
+    quote(generic),
+    quote(class),
+    quote(method)
+  )
+
+  eval(call, envir = environment())
+}
+
+# This Ark API for the Ark version is not stable, so we carefully bail
+# and return `FALSE` if anything looks wrong
```

Comment:
I wonder if returning `TRUE` could be the safer option here. Is there an _existing_ ark version that doesn't have this API? If this ever breaks, we know that it's after the current version, and the other entry points still might exist (and are safeguarded). What happens in the worst case? Should we spice this up with a warning for easier debuggability?